tests/charts/schema_tests.py

```diff
@@ -39,27 +40,27 @@ def test_query_context_limit_and_offset(self):
         # Use defaults
         payload["queries"][0].pop("row_limit", None)
         payload["queries"][0].pop("row_offset", None)
-        query_context, errors = load_query_context(payload)
-        self.assertEqual(errors, {})
+        query_context = load_query_context(payload)
         query_object = query_context.queries[0]
         self.assertEqual(query_object.row_limit, app.config["ROW_LIMIT"])
         self.assertEqual(query_object.row_offset, 0)
 
         # Valid limit and offset
         payload["queries"][0]["row_limit"] = 100
         payload["queries"][0]["row_offset"] = 200
-        query_context, errors = ChartDataQueryContextSchema().load(payload)
-        self.assertEqual(errors, {})
+        query_context = ChartDataQueryContextSchema().load(payload)
         query_object = query_context.queries[0]
         self.assertEqual(query_object.row_limit, 100)
         self.assertEqual(query_object.row_offset, 200)
 
         # too low limit and offset
         payload["queries"][0]["row_limit"] = 0
         payload["queries"][0]["row_offset"] = -1
-        query_context, errors = ChartDataQueryContextSchema().load(payload)
-        self.assertIn("row_limit", errors["queries"][0])
-        self.assertIn("row_offset", errors["queries"][0])
+        try:
+            _ = ChartDataQueryContextSchema().load(payload)
+        except ValidationError as errors:
+            self.assertIn("row_limit", errors.messages["queries"][0])
+            self.assertIn("row_offset", errors.messages["queries"][0])
```

Comment:
If this doesn't raise, won't the assertions just be skipped?

Reply:
good catch!

Reply:
fixed